src/solvers/NLPobjPerFlux.m

```diff
@@ -11,15 +11,13 @@
 % Prob          NLP problem structure
 %
 %OUTPUT
-% value         Objective flux / v.^2
+% value         Objective flux / sum(v.^2)^
 %
 % Markus Herrgard 12/7/07
 %
 % c wasn't defined as written so added Prob as input to define c from the
 % model by Daniel Zielinski 3/19/10
 
-model = Prob.user.model;
+c = Prob.objArguments{1};
 
-c = model.c == 1;
-
-value = -sum(c.*fluxVector)/sum(fluxVector.^2);
+value = sum(c.*fluxVector)/sum(fluxVector.^2);

```

Comment:
Ok.... 
solveCobraNLP expects a minimisation (according to the documentation). Simultaneously optimizeCbModelNLP allows to set whether its a minimsation or maximisation.
Question: should we modify solveCobraNLP to allow both min and maximisations (Wouldn't be so difficult), or should we remove the option from optimzeCbModelNLP? The problem is, that we have function names which are being called (which are expected to be provided as strings, which makes a prior +/- addition impossible.